broccoli-image-writer.js

```diff
@@ -74,9 +74,15 @@ ImageResizer.prototype.build = function () {
  */
 ImageResizer.prototype.generateImages = function (file, sourcePath, destinationPath) {
   let promises = [];
+  let gmImage = gm(path.join(sourcePath, file));
+  let imageInfos = Q.all([
+    Q.ninvoke(gmImage, 'format'),
+    Q.ninvoke(gmImage, 'color'),
+    Q.ninvoke(gmImage, 'size')
+  ]);
   this.image_options.supportedWidths.forEach((width) => {
     promises.push(() => {
-      return this.generateImage(file, sourcePath, destinationPath, width)
+      return this.generateImage(file, sourcePath, destinationPath, width, imageInfos)
```

Comment:
I think you could as well write it like this:
```
return imageInfos.then((infos) => this.generateImage(file, sourcePath, destinationPath, width, infos))
```

So the argument to `generateImage` is the real data, not a promise. Maybe makes the `generateImage` API a bit more simple?